WordPress/playground-tools
# Playground block: Support remote files

## What?

Enables providing remote URLs as filenames in the Playground Block's code editor:

![CleanShot 2024-05-22 at 18 19 40@2x](https://github.com/WordPress/playground-tools/assets/205419/a357b230-90ff-47e1-a78a-04971f4ab4e0)

With this change, you can write an interactive WordPress tutorial using source code hosted on GitHub. It's easier to manage, preview, clone, and fork. While the remote URL is a source of truth for the block, it still stores a copy of the code in block attributes.

## Testing Instructions

1. Insert the Playground block.
2. Create a new file, paste this URL as its name: https://raw.githubusercontent.com/WordPress/block-development-examples/trunk/plugins/data-basics-59c8f8/_playground/blueprint.json
3. Confirm a new file was created with the "link" icon at the top.
4. Try changing its content, saving the post, and refreshing the page. It should have no effect.
5. Confirm regular files still work as they used to.

cc @juanmaguitar @ryanwelcher
